src/_pytest/python.py

```diff
@@ -1126,9 +1126,23 @@ def parametrize(
         # Create the new calls: if we are parametrize() multiple times (by applying the decorator
         # more than once) then we accumulate those calls generating the cartesian product
         # of all calls.
+        # If we parametrize the test using values that contained nested pytest.param objects,
+        # then we unpack them values and merge marks.
         newcalls = []
         for callspec in self._calls or [CallSpec2()]:
             for param_index, (param_id, param_set) in enumerate(zip(ids, parameters)):
+                normalized_values = []
```

Comment:
I would prefer if we move this to a free function and add specific unittests for it.